ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java

```diff
@@ -73,9 +73,21 @@ private void generateCommands() throws IOException, TemplateException {
         }
 
         if (ConverterUtil.hasContent(spec.getTypes())) {
+            // Generate EnumConstant Types, which Element types can be dependant on
+            // TODO Determine if dependency matters at code generation time... probably not.
             for (final Ds3Type typeEntry : spec.getTypes().values()) {
-                generateType(typeEntry);
-                generateTypeMatcher(typeEntry);
+                if (typeEntry.getElements().isEmpty()) {
+                    generateTypeEnumConstant(typeEntry);
+                    generateTypeEnumConstantMatcher(typeEntry);
+                }
+            }
+
+            // Generate Element Types
+            for (final Ds3Type typeEntry : spec.getTypes().values()) {
```

Comment:
Can this for loop be combined with the one above, and just make the generation of element types the else condition to the if above?